jcheck/src/main/java/org/openjdk/skara/jcheck/CopyrightConfiguration.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.skara.jcheck;
+
+import org.openjdk.skara.ini.Section;
+
+public class CopyrightConfiguration {
+    static final CopyrightConfiguration DEFAULT =
+            new CopyrightConfiguration(".*\\.cpp|.*\\.hpp|.*\\.c|.*\\.h|.*\\.java");
```

Comment:
I would like to see more filetypes included by default. At least `Makefile`, `*.gmk`, `configure`, `*.m4`, `*.md` and `*.properties`. Probably also `*.cc`, `*.hh` and `*.m`.

We could also consider not defining defaults here and have it all in .jcheck/conf for better visibility.

Reply:
Ah, I didn't think about the default filetypes, just copied from the default of whitespaceConfiguration. I was thinking we can configure it in .jcheck/conf later like the configuration for  whitespace check https://github.com/openjdk/jdk/blob/41c8971cbcc1d281ffae907c952637ebff945ebb/.jcheck/conf#L19

Reply:
I definitely agree that file patterns that should be checked should be defined in .jcheck/conf.

Reply:
I also think it would be better to not have any defaults and just define the patterns in `.jcheck/conf`

Reply:
Yes, I discussed with Erik offline yesterday, and we agreed on removing the default setting for this check.